src/JSInterop/Microsoft.JSInterop/src/JSRuntimeBase.cs

```diff
@@ -26,6 +25,11 @@ public abstract class JSRuntimeBase : IJSRuntime
 
         internal DotNetObjectRefManager ObjectRefManager { get; } = new DotNetObjectRefManager();
 
+        /// <summary>
+        /// Gets or sets the default timeout for asynchronous JavaScript calls.
+        /// </summary>
+        protected TimeSpan? DefaultAsyncCallTimeout { get; set; }
```

Comment:
RemoteJSRuntime will set this value based on circuit options.

Reply:
Its opt-in so that there is no perf impact in blazor client-side.

Reply:
I think it's smart to have the timeout be optional, I also think we'll want the same behavior in both places for consistency (unless people are actually hitting the timeout in real scenarios).

Reply:
Why do you think its useful to have a timeout on the client?

Reply:
When one environment is more restrictive than the other, people will write components that only function in the less restrictive environment. We've seen this already with the sync-context + dispatching.

Reply:
A default 60-second timeout, even in client-side Blazor, sounds like the right thing to do to me too.

There is some risk that people will have a bad time with it, e.g., they are waiting for user input in some 3rd-party JS UI such as a dialog, and during development and testing they always click the button in time. But then in production they find real users are getting errors because they are responding to the prompt too slowly. The developer would need to know they have to disable the timeout or pick a longer value in this case.

However I'm a little more comfortable with that risk, than with the other risk that they assume it's no timeout by default and are unable to port their components between the two environments.

Reply:
Hmm, should we add a cancellation token parameter and apply the timeout only when a cancellation token is not explicitly provided?

I would be more comfortable with that solution as that would give users the ability to configure the timeout on a per-call to match their expectations. For example, in the case you bring up, a user might choose to pass a custom cancellation token to wait indefinitely or until something else happens.

We will need to make the timeout property public I believe if we want it to be configurable in the WebAssembly case. And I would still allow users to set it to null if they want to disable it alltogether.

**So to summarize my proposal:**
* Add cancellation token parameter users can pass to explicitly control when an operation gets cancelled.
* Make the time-out property public so that it can be configured in WebAssembly and have it default to 60 seconds.
  * Alternatively we can keep the property protected and have it be configured indirectly through other mechanisms.
    * On SSB through circuit options.
    * On CSB through a static property.
  * @SteveSandersonMS What pattern do you suggest to configure this in WebAssembly. Startup code?
* Allow for the property to be set to null to disable time-outs.

Reply:
I agree, cancellation token does sound better!

> What pattern do you suggest to configure this in WebAssembly. Startup code?

I'd have a look at whether it's easy enough to use `services.Configure` and `Options<T>`, where `T` is some new type called something like `WebAssemblyJSRuntimeOptions`.

If that turns out to be inconvenient or bloats the app size by pulling in loads of options-related functionality, a rough but acceptable-in-the-short-term option would be having `Microsoft.AspNetCore.Blazor` contain an extension method on `IJSRuntime` called `SetDefaultTimeout` that writes to a `static` property on the internal `WebAssemblyJSRuntime` class. If we take this approach, it's something we'd want to improve in the future when formalizing the approach to options/config in client-side Blazor.

Reply:
I don't like this name as I would prefer something more explicit or less, like DefaultTimeout, I don't feel it conveys enough information to know what you are setting a timeout for, That said, I'm not a native english speaker so I've switched it to your suggestion.